engine/apps/public_api/serializers/integrations.py

```diff
@@ -166,30 +169,155 @@ def validate_templates(self, templates):
+        """
+        _correct_validated_data_for_legacy_template reshapes validated data to store them.
+        It converts data from "templates" dict to db fields, which were used before messaging backends.
+        Example:
+            {
+              "slack": {
+                "title": Hello
+              }
+            }
+            Will be converted to
+
+            slack_title_template=Hello
+        """
+        if type(template_from_request) is str:  # if it's plain template: {"resolve_signal": "resolve me"}
+            try:
+                validated_data[PUBLIC_TEMPLATES_FIELDS[template_backend_name]] = template_from_request
+            except KeyError:
+                print(3)
+                raise BadRequest(detail="Invalid template data")
+        elif type(template_from_request) is dict:  # if it's nested template: {slack: {"title": "some title"}}
+            for attr, template in template_from_request.items():
                 try:
-                    template = templates[backend_id.lower()][field]
+                    validated_data[PUBLIC_TEMPLATES_FIELDS[template_backend_name][attr]] = template
                 except KeyError:
-                    continue
+                    print(2)
+                    raise BadRequest(detail="Invalid template data")
+        elif template_from_request is None:
+            # if it's we receive None, it's needed to set template to default value
+            try:
+                template_to_set_to_default = PUBLIC_TEMPLATES_FIELDS[template_backend_name]
+                if type(template_to_set_to_default) is str:
+                    # if we receive None for plain template just set it to None
+                    validated_data[PUBLIC_TEMPLATES_FIELDS[template_backend_name]] = None
+                elif type(template_to_set_to_default) is dict:
+                    # if we receive None for nested template set all it's fields to None
+                    for key in template_to_set_to_default.keys():
+                        validated_data[PUBLIC_TEMPLATES_FIELDS[template_backend_name][key]] = None
+            except KeyError:
+                print(4)
+                raise BadRequest(detail="Invalid template data")
+        return validated_data
+
+    def _correct_validated_data_for_messaging_backends_templates(self, validated_data):
+        """
+        _correct_validated_data_for_messaging_backends_templates reshapes validated data to store them.
+        It converts data from "templates" dict to messaging_backends_templates field format.
+        Example:
+            {
+              "msteams": {
+                "title": Hello
+              }
+            }
+            Will be converted to
+
+            messaging_backends={"MSTEAMS": {"title": "Hello"},
+        """
+        templates_data_from_request = validated_data.get("templates", {})
+
+        messaging_backends_templates = self.instance.messaging_backends_templates if self.instance else {}
```

Comment:
`messaging_backends_templates` field is null by default, so it probably makes sense to always do `messaging_backends_templates or {}`

Reply:
Thanks, I thought it will be dict by default